crud/readview.lua

```diff
@@ -0,0 +1,613 @@
+        after = '?table|cdata',
+        first = '?number',
+        batch_size = '?number',
+        bucket_id = '?number|cdata',
+        force_map_call = '?boolean',
+        fields = '?table',
+        fullscan = '?boolean',
+        fetch_latest_metadata = '?boolean',
+
+        mode = '?vshard_call_mode',
+        prefer_replica = '?boolean',
+        balance = '?boolean',
+        timeout = '?number',
+
+        vshard_router = '?string|table',
+
+        yield_every = '?number',
+    })
+
+
+    if opts.first ~= nil and opts.first < 0 then
+        if opts.after == nil then
+            return nil, ReadviewError:new("Negative first should be specified only with after option")
+        end
+    end
+
+    local iterator_opts = {
+        after = opts.after,
+        first = opts.first,
+        batch_size = opts.batch_size,
+        bucket_id = opts.bucket_id,
+        force_map_call = opts.force_map_call,
+        field_names = opts.fields,
+        yield_every = opts.yield_every,
+        call_opts = {
+            mode = opts.mode,
+            prefer_replica = opts.prefer_replica,
+            balance = opts.balance,
+            timeout = opts.timeout,
+            fetch_latest_metadata = opts.fetch_latest_metadata,
+        },
+    }
+    local iter, err = schema.wrap_func_reload(
+            vshard_router, build_select_iterator, space_name, readview_uuid, user_conditions, iterator_opts
+    )
+    if err ~= nil then
+        return nil, err
+    end
+    common.check_select_safety(space_name, iter.plan, opts)
+
+    local tuples = {}
+
+    local count = 0
+    local first = opts.first and math.abs(opts.first)
+    for _, tuple in iter.merger:pairs() do
+        if first ~= nil and count >= first then
+            break
+        end
+
+        table.insert(tuples, tuple)
+        count = count + 1
+    end
+
+    if opts.first ~= nil and opts.first < 0 then
+        utils.reverse_inplace(tuples)
+    end
+    if opts.fetch_latest_metadata then
+        -- This option is temporary and is related to [1], [2].
+        -- [1] https://github.com/tarantool/crud/issues/236
+       -- [2] https://github.com/tarantool/crud/issues/361
+        local storages_info = fiber.self().storage.storages_info_on_select
+        iter = utils.fetch_latest_metadata_for_select(space_name, vshard_router, opts,
+                                                     storages_info, iter)
+    end
+
+    -- filter space format by plan.field_names (user defined fields + primary key + scan key)
+    -- to pass it user as metadata
+    local filtered_space_format, err = utils.get_fields_format(iter.space:format(), iter.plan.field_names)
+    if err ~= nil then
+        return nil, err
+    end
+
+
+    return {
+        metadata = table.copy(filtered_space_format),
+        rows = tuples,
+    }
+end
+
+local Readview_obj = {}
+Readview_obj.__index = Readview_obj
+
+function Readview_obj:select(space_name, user_conditions, opts)
```

Comment:
`:select` and `:pairs` should be wrapped into `stats.wrap(func, stats.op.SELECT)` and stats.wrap(func,  stats.op.SELECT, {pairs = true})

Reply:
added